### PR TITLE
fix(Autocomplete): focus input after clicking on adornment

### PR DESCRIPTION
**Pull request by OndrejTuma**
[FX-3119]

### Description

Currently, when clicking on [Autocomplete adornments](https://picasso.toptal.net/?path=/story/forms-autocomplete--autocomplete#with-adornments), the Input does not focus, leading to opened Select that does not close when clicking outside of Autocomplete. This bug also appears in TagSelector, where values (Tags) are displayed as adornments. 

There are two ways to handle this issue:

**1. Do not open Select when clicking on adornments**

This has a drawback for TagSelector, because values can take up most of the space in Autocomplete, leaving the Input that opens the Select hard to click on. It also reduces UX

**2. Trigger focus on Input when Select is open**

This requires having a reference to Input. Possible drawback is when user defines a reference to Autocomplete as a function - this way we can't reach the input to trigger focus. But that should be edge case, since standard way to create refs in React is through `useRef` hook.

I went for solution 2., although I'm open to better suggestions 🙇 

### How to test

- clicking on [TagSelector's values](https://picasso.toptal.net/fx-3119-tagselector-input-doesn-t-get-focused-when-click-on-a-selected-option/?path=/story/forms-tagselector--tagselector#initially-set-value) should focus the Input

### Before

https://user-images.githubusercontent.com/22159416/210854017-73dc085a-6517-4117-81a5-9d08569c302f.mov

### After

https://user-images.githubusercontent.com/22159416/210854046-889a9660-9caa-460b-84d5-44134e98e7e8.mov

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Annotate all `props` in component with documentation
- Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
4. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
5. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3119]: https://toptal-core.atlassian.net/browse/FX-3119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dmaklygin — COMMENTED** [review]
checked the PR. Works as expected.
I provided an idea how to handle input focus, but maybe Ondra's solution is more common

**Comment by OndrejTuma** [comment]
> checked the PR. Works as expected. I provided an idea how to handle input focus, but maybe Ondra's solution is more common

So I checked the solution @dmaklygin  suggested - adding `onClick` to `TagSelectorLabel`. 
It works well for TagSelector, but it does not solve clicking on adornments in Autocomplete. Also, there's a possibility to pass custom function to render a TagSelector label, in which case the onClick would be applied.

**Review by LashaJini — APPROVED** [review]
Tested and works as expected :+1:

File: packages/shared/src/utils/is-forward-ref.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const isForwardRef = (Component: any) =>
+  typeof Component === 'object' &&
+  Component !== null &&
+  Component.$$typeof === Symbol.for('react.forward_ref')
```
Comment (OndrejTuma) [inline packages/shared/src/utils/is-forward-ref.ts:4]: I decided to use custom implementation because [react-is](https://github.com/facebook/react/blob/main/packages/react-is/src/ReactIs.js#L124-L126) requires passing instance of component.

**Review by TomasSlama — APPROVED** [review]
Tested, well done 👍

**Review by dmaklygin — APPROVED** [review]
Good work!

**Review by augustobmoura — APPROVED** [review]
Looks great @OndrejTuma! :+1: Good job!
